libsql-sqlite3/src/pager.c

```diff
@@ -4140,9 +4140,10 @@ int sqlite3PagerClose(Pager *pPager, sqlite3 *db){
       a = pTmp;
     }
     if (pagerUseWal(pPager)) {
-      pPager->create_wal->xClose(pPager->create_wal->pData, pPager->wal.pData, db, pPager->walSyncFlags, pPager->pageSize,a);
+      pPager->create_wal->ref.xClose(pPager->create_wal->ref.pData, pPager->wal.pData, db, pPager->walSyncFlags, pPager->pageSize,a);
+      memset((void*)&(pPager->wal), 0, sizeof(libsql_wal));
     }
-    memset((void*)&(pPager->wal), 0, sizeof(libsql_wal));
+    destroy_create_wal(&pPager->create_wal);
```

Comment:
that one should only be called if `pagerUseWal(pPager)`, right? So it should be moved one line up

Reply:
The create wal is always there, but a wal is only created if the wal journaling mode is enabled. So we always need to destroy it.